model/Resource/Service/ResourceIndexer.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ *
+ * @author Gabriel Felipe Soares <gabriel.felipe.soares@taotesting.com>
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\model\Resource\Service;
+
+use core_kernel_classes_Resource;
+use oat\tao\model\search\tasks\UpdateResourceInIndex;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\taoAdvancedSearch\model\Index\Service\IndexerInterface;
+
+class ResourceIndexer implements IndexerInterface
+{
+    /** @var QueueDispatcherInterface */
+    private $queueDispatcher;
+
+    public function __construct(QueueDispatcherInterface $queueDispatcher)
+    {
+        $this->queueDispatcher = $queueDispatcher;
+    }
+
+    public function addIndex($resource): void
```

Comment:
suggestion: Maybe `addToIndex` or `indexResource` is a bit more clear? `addIndex` may sound as adding a new ES index to some list of indexes the resource is indexed in.

```suggestion
    public function addToIndex($resource): void
```

Reply:
Got it, I was looking at the interface at `oat\tao\elasticsearch` instead of `oat\taoAdvancedSearch\model\Index\Service` :man_facepalming: